### PR TITLE
finished main functionality and added testing 

### PR DESCRIPTION
**Pull request by hiimbex**
Added the main functionality to the bot, more documentation to come!

cc/ @kytrinyx @lindseyb @dmleong @ihavenotea @bkeepers

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
 module.exports = robot => {
     robot.on('pull_request.opened', receive);
     robot.on('issues.opened', receive);
     async function receive(context) {
-        const title = context.payload.issue.title;
-        const body = context.payload.issue.body;
-        // robot.log('Title: ', title, 'Body: ', body);
-
-        // Add options for if there is a default title?
+        let title, body, number, bodies, badTitle;
```
Comment (dmleong) [inline index.js]: Pardon the review pre ☕️  but what's the difference between `body` and `bodies`?

File: index.js [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
+        if (context.payload.pull_request) {
+            ({ title, body, number } = context.payload.pull_request);
+        } else {
+            ({ title, body, number } = context.payload.issue);
+        }
 
-        if (!body) {
-            const options = context.repo({path: '.github/request-info.md'});
+        try {
```
Comment (dmleong) [inline index.js]: Nice catch/try here 😄

File: index.js [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
+            const options = context.repo({path: '.github/config.yml'});
             const response = await context.github.repos.getContent(options);
-            const template = Buffer.from(response.data.content, 'base64').toString();
+            bodies = yaml.load(Buffer.from(response.data.content, 'base64').toString()) || {};
+        } catch (err) {
+            if (err.code !== 404) throw err;
+        }
 
-            context.github.issues.createComment(context.issue({body: template}));
+        if (bodies) {
```
Comment (dmleong) [inline index.js]: It's a little confusing to have something related to `bodies` here and then `body` in the lines below. Maybe breaking these into their own functions might help with readability?

File: index.js [hunk 4]
```diff
@@ -1,19 +1,39 @@
...
+        }
+        if (!body || badTitle) {
+            context.github.issues.createComment(context.issue({body: bodies.requestInfoReplyComment}));
+            if (bodies.requestInfoLabelToAdd) {
+                // Add label if there is one listed in the yaml file
+                const repo = context.payload.repository;
+                context.github.issues.addLabels({
+                    owner: repo.owner.login,
+                    repo: repo.name,
+                    number,
```
Comment (dmleong) [inline index.js]: Any reason this isn't `number: number`?

File: test/events/failEvent.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "event": "issues", 
+    "payload": {
+      "action": "opened",
+      "issue": {
+        "body": "Fix your broken thigns!",
+        "title": "ur thing is broken fix it!!",
+        "user": {
+          "login": "hiimbex"
+        },
+        "labels": []
```
Comment (dmleong) [inline test/events/failEvent.json]: Pardon my ignorance of the API, but will this fail if you omit the `labels` part of the payload altogether? Does the API give you an empty array by default?

File: index.js (hunk 1 above)
Comment (hiimbex) [inline index.js]: `body` is the actually body of text when opening an issue or a PR while as `bodies` was supposed to be renamed to `config` but I forgot :)

File: index.js (hunk 4 above)
Comment (hiimbex) [inline index.js]: The linter expects property shorthand: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer

File: index.js (hunk 3 above)
Comment (hiimbex) [inline index.js]: given my above comment, does renaming to `config` provide enough clarification/readability? Or do you think separate functions would help?

File: test/events/failEvent.json (hunk 5 above)
Comment (hiimbex) [inline test/events/failEvent.json]: So these are just fake payloads that I used for testing; however, I double checked and it's not actually necessary for the mocking, so I'll remove them 👍 But I believe I copied them and that the API does return an empty list if there are no labels, but I'm not positive and would have to double check

File: index.js (hunk 3 above)
Comment (dmleong) [inline index.js]: Clarifying comment helped a lot, thanks! Looks fine as is

**Pull request merged**